repository/src/lib.rs

```diff
@@ -316,11 +316,93 @@ impl<T: RawRepository> DistributedRepository<T> {
         unimplemented!()
     }
 
-    /// Creates an agenda commit on top of the `work` branch.
+    /// Creates an extra-agenda transaction commit on top of the `work` branch.
     pub async fn create_extra_agenda_transaction(
         &mut self,
-        _transaction: &ExtraAgendaTransaction,
+        transaction: &ExtraAgendaTransaction,
     ) -> Result<CommitHash, Error> {
-        unimplemented!()
+        let work_commit = self.raw.locate_branch(WORK_BRANCH_NAME.into()).await?;
+        let last_header_commit = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+
+        // Check if the `work` branch is rebased on top of the `finalized` branch.
+        if self
+            .raw
+            .find_merge_base(last_header_commit, work_commit)
+            .await?
+            != last_header_commit
+        {
+            return Err(anyhow!(
+                "branch {} should be rebased on {}",
+                WORK_BRANCH_NAME,
+                FINALIZED_BRANCH_NAME
+            ));
+        }
+
+        // Construct a commit list starting from the next commit of the last finalized block to the `branch_commit`(the most recent commit of the branch)
+        let ancestor_commits = self.raw.list_ancestors(work_commit, Some(256)).await?;
```

Comment:
Why is this number Some(256), not None

Reply:
Is this because we use Buffered(256) later??

Reply:
It's because it might take long if it goes up to far old commit. 256 is just a random number to prevent it.